var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -462,24 +462,24 @@ def flag_handler(self, name, flags):
 
     def determine_toolset(self, spec):
         toolsets = {
-            "g++": "gcc",
-            "icpc": "intel",
-            "icpx": "intel",
-            "clang++": "clang",
-            "armclang++": "clang",
-            "xlc++": "xlcpp",
-            "xlc++_r": "xlcpp",
-            "pgc++": "pgi",
-            "nvc++": "pgi",
-            "FCC": "clang",
+            "%gcc": "gcc",
+            "%intel": "intel",
+            "%oneapi": "intel",
```

Comment:
I'm not sure if Boost builds with the oneapi compiler. I'll have to double-check.